src/composeappmanager.cc

```diff
@@ -206,13 +208,15 @@ ComposeAppManager::AppsContainer ComposeAppManager::getAppsToUpdate(const Uptane
     if (app_data.empty()) {
       // new app in Target
       apps_to_update.insert(app_pair);
+      apps_and_reasons.push_back(app_pair.first + " (new app in target)");
       LOG_INFO << app_name << " will be installed";
       continue;
     }
 
     if (app_pair.second != app_data["uri"].asString()) {
       // an existing App update
       apps_to_update.insert(app_pair);
+      apps_and_reasons.push_back(app_pair.first + " (new version)");
```

Comment:
The same as above, this is the new target case - in this case a reason of update is a new Target update.